arup-group/a-carbon-tool
# Report v2

Fairly large pr making an update to the report object. Contains a couple of changes:

1. Update to include material carbon totals in the object so that child objects do not need to be pulled down into most pages (no pages require child objects atm, the assessment view page might at some point), reducing load times.
2. Added a "version" property to the object. This is the version of the object that is being used, not the version of the users report. This version number is used to let Act know how to load the report. It allows us to support both this new version of the report object and also the old version (and any new versions that we make).
3. Adds a new version of the model to the report, this version has the carbon values added to the model as new parameters. This allows us to more easily display a heat map on the report view page (which is also a new feature in this pr). Generating this model can take quite a while, so the time to save a report has increased. But I think that the trade-off is worth it.

More info on some of these changes here #180.

User's pre-update reports will not be directly updated as that would have a high chance of causing issues. Instead, any new reports that are created post-update will be made using this version of the report.

To test:
1. Create a new report and check that it loads properly (with the heat map and everything)
2. Edit a v1 report and see if that update works
3. There is meant to be full backwards compatibility with v1 reports, so check that the report view page still works on v1 reports
